csrc/multi_tensor_adam.cu

```diff
@@ -126,6 +126,118 @@ struct AdamFunctor
   }
 };
 
+template<typename T>
+struct AdamCapturableFunctor
+{
+   __device__ __forceinline__ void operator()(
+    int chunk_size,
+    volatile int* noop_gmem,
+    TensorListMetadata<4>& tl,
+    const float beta1,
+    const float beta2,
+    const int* step,
+    const int bias_correction,
+    const float epsilon,
+    const float* lr,
+    adamMode_t mode,
+    const float decay,
+    const float* inv_scale)
```

Comment:
Is this the argument which differentiates this functor from the existing `AdamFunctor`?

Reply:
This functor is different from the original with the following:
1) const int* step - pointer, since it is now being used inside the kernel
2) const int bias_correction - same, see [here](https://github.com/NVIDIA/apex/blob/6588f6b03cf60ba65941d236a19dd5d046e7f90a/csrc/multi_tensor_adam.cu#L256-L261)
3) const float* lr - pointer, to support LR scheduling while captured
4) const float* inv_scale - same